src/components/Selector/MultiSelector.tsx

```diff
@@ -156,13 +156,17 @@ export function MultiSelector<
   );
 }
 
-type SelectedProps<T> = {
+type SelectedProps<T extends ValKey> = {
```

Comment:
no need to narrow this down as this is internal to `Multiselector` that is already bound by `ValKey`

Reply:
No harm in narrowing it down.

Reply:
Why are we suddenly narrowing types of an internal implementation of `Multiselector` which api is already narrowed down for safe use? Do we plan to use `<SelectedProps/>` outside of `Multiselector` ?  I mean, what's even the point of this PR? I've been going here back and forth away from more important task. @SovereignAndrey